roles/postgres/tasks/upgrade_postgres.yml

```diff
@@ -0,0 +1,135 @@
+---
+
+# Upgrade Posgres (Managed Databases only)
+#  * If postgres version is not 12, and not an external postgres instance (when managed_database is yes),
+#    then run this playbook with include_tasks from database_configuration.yml
+#  * Data will be streamed via a pg_dump from the postgres 12 pod to the postgres 13
+#    pod via a pg_restore.
+
+
+- name: Scale down Deployment for migration
+  include_tasks: scale_down_deployment.yml
+
+- name: Delete existing postgres configuration secret
+  k8s:
+    api_version: v1
+    kind: Secret
+    name: "{{ ansible_operator_meta.name }}-postgres-configuration"
+    namespace: "{{ ansible_operator_meta.namespace }}"
+    state: absent
+    wait: yes
+
+- name: Create Database configuration with new -postgres-{{ postgres_version }} hostname
+  k8s:
+    apply: true
+    definition: "{{ lookup('template', 'postgres_upgrade.secret.yaml.j2') }}"
+  no_log: true
+
+- name: Set new database var to be used when configuring app credentials (resources_configuration.yml)
+  set_fact:
+    postgres_host: "{{ ansible_operator_meta.name }}-postgres-{{ postgres_version }}"
+  no_log: true
+
+- name: Create Database if no database is specified
+  k8s:
+    apply: true
+    definition: "{{ lookup('template', 'postgres.yaml.j2') }}"
+    wait: true
+  register: create_statefulset_result
+
+- name: Set postgres label if not defined by user
+  set_fact:
+    postgres_label_selector: "app.kubernetes.io/instance=postgres-{{ postgres_version }}-{{ ansible_operator_meta.name }}"
+  when: postgres_label_selector is not defined
+
+- name: Get new postgres pod information
+  k8s_info:
+    kind: Pod
+    namespace: "{{ ansible_operator_meta.namespace }}"
+    label_selectors:
+      - "{{ postgres_label_selector }}"
+      - app.kubernetes.io/version: '{{ postgres_version }}'
+    field_selectors:
+      - status.phase=Running
+  register: postgres_pod
+  until:
+    - "postgres_pod['resources'] | length"
+    - "postgres_pod['resources'][0]['status']['phase'] == 'Running'"
+    - "postgres_pod['resources'][0]['status']['containerStatuses'][0]['ready'] == true"
+  delay: 5
+  retries: 60
```

Comment:
Nice to have would be capturing a failure status message if this doesn't complete in the 60 retries.

Reply:
I could see us trying to set an entry here  this_pulp['status']['conditions'] to create an entry like this:
```
  status:
    adminPasswordSecret: example-pulp-admin-password
    conditions:
    - lastTransitionTime: "2022-05-17T17:49:58Z"
      reason: Successful
      status: "False"
      type: Successful
```

But have a more useful message for "reason".  

Let's do that as follow-up work though, for now I want to get some working builds.